examples/v1/100-cpu-limit.yaml

```diff
@@ -28,24 +28,25 @@ spec:
           operator: Gt
           values: ["2"]
       nodeClassRef:
-        apiVersion: karpenter.k8s.aws/v1beta1
+        group: karpenter.k8s.aws
         kind: EC2NodeClass
         name: default
   limits:
     cpu: 100
 ---
-apiVersion: karpenter.k8s.aws/v1beta1
+apiVersion: karpenter.k8s.aws/v1
 kind: EC2NodeClass
 metadata:
   name: default
   annotations:
     kubernetes.io/description: "General purpose EC2NodeClass for running Amazon Linux 2 nodes"
 spec:
-  amiFamily: AL2 # Amazon Linux 2
   role: "KarpenterNodeRole-${CLUSTER_NAME}" # replace with your cluster name
   subnetSelectorTerms:
     - tags:
         karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
   securityGroupSelectorTerms:
     - tags:
-        karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
\ No newline at end of file
+        karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  amiSelectorTerms:
+    - alias: al2023@v20240625 # Amazon Linux 2023
```

Comment:
Why pin this specific AMI?

Reply:
This is just an example, no reason we pin this AMI. I wanted to get customers accustomed to pinning their AMIs

Reply:
Do we mention in the docs already that pinning AMIs is something that users should be doing?

Reply:
We recommend pinning throughout our documentation, although I think that's easy for us to do in our docs because we can always change out the value with a couple of bash commands. This one I think is fine to keep as is, but understand it'll go stale in due time.

Reply:
> Do we mention in the docs already that pinning AMIs is something that users should be doing?

https://karpenter.sh/docs/tasks/managing-amis/#choosing-ami-options